python/lsst/analysis/ap/zooniverseCutouts.py

```diff
@@ -88,39 +117,35 @@ def run(self, data, butler, outputPath):
         butler : `lsst.daf.butler.Butler`
             The butler connection to use to load the data; create it with the
             collections you wish to load images from.
-        outputPath : `str`
-            The path to write the output to; manifest goes here, while the
-            images themselves go into ``outputPath/images/``.
-        """
-        result = self.write_images(data, butler, outputPath)
+        njobs : `int`, optional
+            Number of multiprocessing jobs to make cutouts with; default of 0
+            means don't use multiprocessing at all.
 
-        if self.config.urlRoot is not None:
-            manifest = self.make_manifest(result)
-            manifest.to_csv(os.path.join(outputPath, "manifest.csv"), index=False)
-        else:
-            self.log.warning("No urlRoot provided, so no manifest file written.")
-
-        self.log.info("Wrote %d images to %s", len(result), outputPath)
+        Returns
+        -------
+        source_ids : `list`
+            DiaSourceIds of cutout images that were generated.
+        """
+        result = self.write_images(data, butler, njobs=njobs)
+        self.write_manifest(result)
+        self.log.info("Wrote %d images to %s", len(result), self._output_path)
+        return result
 
-    @staticmethod
-    def _make_path(id, base_path):
-        """Return a URL or file path for this source.
+    def write_manifest(self, sources):
+        """Save a Zooniverse manifest attaching image URLs to source ids.
 
         Parameters
         ----------
-        id : `int`
-            The source id of the source.
-        base_path : `str`
-            Base URL or directory path, with no ending ``/``.
-
-        Returns
-        -------
-        path : `str`
-            Formatted URL or path.
+        sources : `list` [`int`]
+            The diaSourceIds of the sources that had cutouts succesfully made.
         """
-        return f"{base_path}/images/{id}.png"
+        if self.config.url_root is not None:
+            manifest = self.make_manifest(sources)
+            manifest.to_csv(os.path.join(self._output_path, "manifest.csv"), index=False)
```

Comment:
I changed it from `warning` to `info` and moved it out of `run`, because now I think that most of the use cases are going to be *not* writing the manifest (i.e. our own debugging of AP runs, where all we care about are the cutouts themselves). Generating the manifest after the fact from the files on disk is always possible, so it's not a huge loss, and a `warning` might make people wonder if they broke something when they don't actually care.

It's an unfortunate fact of how this code evolved over time, from something specifically for Zooniverse, to something we'll use for debugging AP.

Reply:
Ah, OK, I see 👍